crates/admin/src/cluster_controller/replicated_loglet_controller.rs

```diff
@@ -0,0 +1,656 @@
+                            self.config.replication.num_copies() as usize
+                        );
+
+                        let sequencer = cluster_state
+                            .healthy_workers
+                            .values()
+                            .choose(&mut rand::thread_rng())
+                            .cloned()
+                            .expect("we have at least one healthy worker");
+
+                        let loglet_id = rand::thread_rng().next_u64(); // should we make this a proper Restate ID?
+                        let params = ReplicatedLogletParams {
+                            loglet_id: ReplicatedLogletId::new(loglet_id), // todo: should this go through metadata to ensure global uniqueness?
+                            nodeset: node_set.clone(),
+                            replication: self.config.replication.clone(),
+                            sequencer,
+                            write_set: None,
+                        };
+
+                        updated_plan.insert_loglet(TargetLogletState {
+                            log_id: *log_id,
+                            replication: self.config.replication.clone(),
+                            loglet_id: ReplicatedLogletId::new(loglet_id),
+                            params: LogletParams::from(
+                                params.serialize().expect("can always serialize params"),
+                            ),
+                            segment_index: SegmentIndex::default(),
+                            node_set,
+                            sequencer: Some(sequencer),
+                        });
+
+                        // todo: what if we previously proposed this action? how do we avoid doing this unnecessarily?
+                        effects.push(LogletEffect::AddLogletSegment);
+                    }
+
+                    Some(target_state) => {
+                        // Check if loglet configuration requires any remediating actions
+                        let mut nodes_config = NodesConfiguration::default(); // todo: this should come from observed state
+                        cluster_state
+                            .healthy_workers
+                            .iter()
+                            .to_owned()
+                            .for_each(|(node_id, _)| {
+                                nodes_config.upsert_node(NodeConfig::new(
+                                    format!("node-{}", node_id),
+                                    cluster_state.healthy_workers[node_id],
+                                    format!("unix:/tmp/my_socket-{}", node_id).parse().unwrap(),
+                                    Role::LogServer.into(),
+                                    LogServerConfig {
+                                        storage_state: cluster_state.log_server_states[node_id],
+                                    },
+                                ));
+                            });
```

Comment:
The cluster controller won't decide which node runs as a log server. That's something that the node is started with or not.